tests/test_om_prior.py

```diff
@@ -1,32 +1,175 @@
+import datetime
+import os
 import subprocess
+import sys
+from pathlib import Path
 
+import numpy as np
+import pandas as pd
 import pytest
+import requests
 import xarray as xr
-import os
+from openmethane_prior.layers.omGFASEmis import downloadGFAS
+from openmethane_prior.omInputs import livestockDataPath, sectoralEmissionsPath
+from openmethane_prior.omUtils import getenv, secsPerYear
+
+root_path = Path(__file__).parent.parent
+sys.path.insert(1, os.path.join(root_path, "scripts"))
+from omDownloadInputs import download_input_files, downloads, remote, sectoralEmissionsPath
+
+
+@pytest.fixture(scope="session")
+def output_domain_file(root_dir,
+                       input_files
+                       ) :
+    subprocess.run(["python", os.path.join(root_dir, "scripts/omCreateDomainInfo.py")], check=True)
+
+    subprocess.run(["python", os.path.join(root_dir, "scripts/omPrior.py"), "2022-07-01", "2022-07-02"], check=True)
+
+    # Generated by scripts/omCreateDomainInfo.py
+    filepath_in_domain = os.path.join(root_dir, "inputs/om-domain-info.nc")
+
+    # Generated by scripts/omPrior.py
+    filepath_out_domain = os.path.join(root_dir, "outputs/out-om-domain-info.nc")
```

Comment:
Can we add a new test that verifies that this output file has the same lat/lons that are in the CRO/DOT files. `filepath_in_domain` should have the same dimensions as well.

Reply:
Is it possible that `GRIDCRO2D_1` and `GRIDDOT2D_1` have different dimensions?  

In this test I compare `COL` and `ROW` of `GRIDCRO2D_1` and `GRIDDOT2D_1` with `inputs/om-domain-info.nc`. 

The number of columns and rows  in `GRIDCRO2D_1` is one below the number of columns and rows in `GRIDDOT2D_1`:

`GRIDCRO2D_1` contains the attributes `ROW: 430, COL: 454`

and 

`GRIDDOT2D_1` contains the attributes `ROW: 431, COL: 455`.

In `om-domain-info.nc` there are the attributes `ROW: 430` and `ROW_D: 431`.
So I can compare `ROW` with  `GRIDCRO2D_1` and `ROW_D` with `GRIDDOT2D_1`?


The test that I am adding looks like this. It's passing, but I am not sure if the test makes sense.

```python
def test_007_compare_in_domain_with_cro_dot_files(input_domain_xr, cro_xr, dot_xr):

    assert dot_xr.NCOLS == input_domain_xr.COL_D.size
    assert dot_xr.NROWS == input_domain_xr.ROW_D.size

    assert cro_xr.NCOLS == input_domain_xr.COL.size
    assert cro_xr.NROWS == input_domain_xr.ROW.size
```

Reply:
WRF and CMAQ use a staggered grid (https://amps-backup.ucar.edu/information/configuration/wrf_grid_structure.html) where depending on the quantity that you are measuring the grid points are slightly different. Things like temperature/pressure are defined at the center of a grid cell, where as other quantities such as wind are defined at the edges of a grid cell. If you count the number of edges in a grid along a x or y it is one larger than the number of cells in x or y.

The CRO/DOT files are generated by MCIP (?) from the WRF domain and define the domain used by CMAQ.
The CRO files define the latitude centers where as the DOT files define the corners hence why `ROW_D = ROW + 1`. I don't know why they are named as they are 🤷 

The CMAQ domain is a little smaller than the WRF domain because the 5 grid cells around the edge of the domain are excluded. I can talk you through why that is another day.

Reply:
According to this [website](https://www.cmascenter.org/ioapi/documentation/all_versions/html/GRIDS.html), CRO is short for cross and dot. 

An image of the CMAQ grid: https://www.cmascenter.org/ioapi/documentation/all_versions/html/GRIDS.jpg
and the interaction between the grid and "thickened boundary" https://www.cmascenter.org/ioapi/documentation/all_versions/html/THKBDY.jpg. The outside is the WRF domain, but the inner cells labelled "main grid" is the CMAQ grid

Reply:
Ok thanks. I think I have understood it to some degree. 

Is the difference between CMAQ domain and WRF domain (the thickened boundary) unrelated to mass grid / staggered grid problem? And it wouldn't be relevant to the tests here anyway, would it?

I've added the tests 007 and 008.

Reply:
Correct. It is unrelated. If you compared the grid sizes in the raw input domain `geo_em.d01.nc` you would see that they don't align with the CMAQ grid